pkg/controller/vsphere/reconciler_test.go

```diff
@@ -1247,52 +1259,118 @@ func TestDelete(t *testing.T) {
 		},
 	}
 
+	nodeName := "somenodename"
+
+	getMachineWithStatus := func(status machinev1.MachineStatus) *machinev1.Machine {
```

Comment:
You need to pass t in here as well
```suggestion
	getMachineWithStatus := func(t *testing.T, status machinev1.MachineStatus) *machinev1.Machine {
```

Reply:
Yup, missed that. Fixed.

Reply:
Also squashed that.